python/plugins/roll.py

```diff
@@ -0,0 +1,42 @@
+import re
+import random
+from plugins.BasePlugin import BasePlugin
+
+class RollPlugin(BasePlugin):
+
+    def parseMessage(self, username, serverPart, message, isPrivateMessage):
+        rollPattern = re.compile(self.getBotName() + ": roll " +  r"((\d+)?d(\d+)([+|-])?(\d+)?)")
+        rollMatch = rollPattern.match(str(message))
+        if(rollMatch):
+            output = "Rolling " + str(rollMatch.group(1)) + ". "
+            multiplier = (rollMatch.group(2))
+            dice = int(rollMatch.group(3))
+            op = rollMatch.group(4)
+            mod = rollMatch.group(5)
+            if(not multiplier):
+                multiplier = 1
+            if(dice < 1):
```

Comment:
I guess you could add in the condition to stop kiwi looping for millions of dice with !roll 1000000d6 or something if you wanted. Whatever you want really, the implementation is just for us anyways. :-)